tests/snappi/qos/files/packet_reorder_helper.py

```diff
@@ -0,0 +1,321 @@
+    port_id = get_dut_port_id(dut_hostname=duthost.hostname,
+                              dut_port=dut_port,
+                              conn_data=conn_data,
+                              fanout_data=fanout_data)
+
+    pytest_assert(port_id is not None,
+                  'Fail to get ID for port {}'.format(dut_port))
+
+    flow_rate_percent = int(FLOW_AGGR_RATE_PERCENT / len(flow_prio_list))
+
+    # Generate traffic config
+    __gen_traffic(duthost=duthost,
+                  testbed_config=testbed_config,
+                  port_config_list=port_config_list,
+                  port_id=port_id,
+                  flow_name=FLOW_NAME,
+                  flow_prio_list=flow_prio_list,
+                  flow_rate_percent=flow_rate_percent,
+                  total_tx_pkts=TOTAL_NUM_PKTS,
+                  larger_pkt_size=LARGER_PKT_SIZE,
+                  smaller_pkt_size=SMALLER_PKT_SIZE,
+                  pkt_step_size=PKT_STEP_SIZE,
+                  prio_dscp_map=prio_dscp_map)
+
+    flows = testbed_config.flows
+    all_flow_names = [flow.name for flow in flows]
+
+    # Run traffic
+    flow_metrics = __run_traffic(api=api,
+                                 config=testbed_config,
+                                 all_flow_names=all_flow_names,
+                                 timeout=EXP_FLOW_DUR_SEC)
+
+    # Verify results i.e. no out of order packets
+    __verify_results(api=api,
+                     flow_metrics=flow_metrics,
+                     exp_rx_pkts=TOTAL_NUM_PKTS)
+
+
+def __gen_traffic(duthost,
+                  testbed_config,
+                  port_config_list,
+                  port_id,
+                  flow_name,
+                  flow_prio_list,
+                  flow_rate_percent,
+                  total_tx_pkts,
+                  larger_pkt_size,
+                  smaller_pkt_size,
+                  pkt_step_size,
+                  prio_dscp_map):
+    """
+    Generate configurations of flows, and device configurations on both the DUT, and ixia device which
+    emulates a neighbor.
+    Args:
+        duthost (Ansible host instance): device under test
+        testbed_config (obj): testbed L1/L2/L3 configuration
+        port_config_list (list): list of port configuration
+        port_id (int): ID of DUT port to test
+        flow_name (str): name of flow
+        flow_prio_list (list): priorities of the flow
+        flow_rate_percent (int): rate percentage for each flow
+        total_tx_pkts (int): total number of packets to transmit
+        larger_pkt_size (int): packet size of larger data flow in bytes
+        smaller_pkt_size (int): packet size of smaller data flow in bytes
+        pkt_step_size (int): packet size step of flow in bytes
+        prio_dscp_map (dict): Priority vs. DSCP map (key = priority).
+    Returns:
+        N/A
+    """
+
+    rx_port_id = port_id
+    tx_port_id_list, _ = select_ports(port_config_list=port_config_list,
```

Comment:
Many to one traffic may cause congestion and packet drops

Reply:
The alternative is all-to-all which is not really needed. I need to modify select_ports to have a "one-to-one" configuration.

Reply:
This doesn't matter since we only take in one tx and rx port using the "next" call in line 143